### PR TITLE
Migrate config files to typescript

### PR DESCRIPTION
**Pull request by ayobuenavista**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "compilerOptions": {
+    "target": "es2018",
+    "module": "commonjs",
+    "strict": true,
+    "esModuleInterop": true,
+    "outDir": "dist",
+    "resolveJsonModule": true,
```
Comment (ducquangkstn) [inline tsconfig.json:8]: pls add "allowJs": true, so that we can import js code in ts file

**Review by ducquangkstn — COMMENTED** [review]
- Pls fix travis
- Write a simple ts test for liquidityMining folder to make sure config works

File: lint.sh [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
 yarn solhint 'contracts/**/*.sol'
 if [ ! -z "$(find test/ -name "*.js")" ];
```
Comment (ducquangkstn) [inline lint.sh:3]: actually a lot of test files are not check linting. It should be
```bash
yarn prettier -c test/*
```

File: hardhat.config.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
+import '@nomiclabs/hardhat-truffle5';
+import '@nomiclabs/hardhat-ethers';
+import '@nomiclabs/hardhat-web3';
+import '@nomiclabs/hardhat-etherscan';
+import 'hardhat-gas-reporter';
+import 'solidity-coverage';
+import '@openzeppelin/hardhat-upgrades';
+import 'hardhat-typechain';
```
Comment (ducquangkstn) [inline hardhat.config.ts:8]: Pls specify which typechain types are generated
https://github.com/KyberNetwork/kyber_bridge_reserves_sc/blob/master/hardhat.config.ts#L60

**Review by ducquangkstn — COMMENTED** [review]
LGTM. overall.

Since we are in a rush, you can remove test sample file (if you are not familiar with writing truffle test with ts)

File: lint.sh (hunk 2 above)
Comment (ayobuenavista) [reply lint.sh:3]: This setting will make it fail in travis, and also since the *.js tests aren't linted. So I changed it to apply to the *.ts files only.

**Pull request merged**